libweb3jsonrpc/UnixSocketServer.cpp

```diff
@@ -111,14 +111,15 @@ void UnixDomainSocketServer::Listen()
 			DEV_GUARDED(x_sockets)
 				m_sockets.insert(connection);
 
-			std::thread handler([this, connection](){ GenerateResponse(connection); });
+			std::thread handler([this, connection](){ GenerateResponse(connection); CloseConnection(connection);});
 			handler.detach();
 		}
 	}
 }
 
 void UnixDomainSocketServer::CloseConnection(int _socket)
 {
+	shutdown(_socket, SHUT_RDWR);
```

Comment:
The top two answers are give some insight: https://stackoverflow.com/questions/4160347/close-vs-shutdown-socket